backend/src/org/commcare/session/SessionInstanceBuilder.java

```diff
@@ -39,7 +39,8 @@ private static void addSessionNavData(TreeElement sessionRoot, SessionFrame fram
      */
     private static void addDatums(TreeElement sessionData, SessionFrame frame) {
         for (StackFrameStep step : frame.getSteps()) {
-            if (SessionFrame.STATE_DATUM_VAL.equals(step.getType())) {
+            if (SessionFrame.STATE_DATUM_VAL.equals(step.getType()) ||
```

Comment:
Similar to above, we have this new type that we can also possibly want to move into the sessionData instance